atomicapp/utils.py

```diff
@@ -378,6 +378,60 @@ def rm_dir(directory):
         distutils.dir_util.remove_tree(directory)
 
     @staticmethod
+    def getUserHome():
+        """
+        Finds the home directory of the user running the application.
+        Uses the SUDO_USER and USER environment variables. If runnning within
+        a container, SUDO_USER and USER varibles must be passed for proper
+        detection as well the root dir /.
+        Ex. docker run -v /:/host -e SUDO_USER -e USER foobar
+        """
+        logger.debug("Finding the users home directory")
+        sudo_user = os.environ.get('SUDO_USER')
+
+        if os.getegid() == 0 and sudo_user is None:
+            user = 'root'
+        elif sudo_user is not None:
+            user = sudo_user
+        else:
+            user = os.environ.get('USER')
+
+        # Check to see if we are running in a container. If we are we will use
+        # the /host path for finding the user directory. We are unable to use
+        # os.path.expanduser due to the passed volume
+        if Utils.areWeContainerized():
+            logger.debug("Atomic App is running in a container")
+            if not os.path.isdir("/host"):
+                raise AtomicAppUtilsException(
+                    "/ host directory must be passed to the Atomic App container.")
+            if user is "root":
+                home = "/host/root"
+            else:
+                home = "/host/home/%s" % user
```

Comment:
This is statically setting the home directory based on the username. This makes an assumption that is unsafe.